R/factorize.R

```diff
@@ -58,23 +70,35 @@ factorize_secuTrial.secuTrialdata <- function(object) {
 # # data.frame method
 # nolint end
 factorize_secuTrial.data.frame <- function(data, cl, form, items) {
+  # character reformatting
   if (!is.character(cl$column)) cl$column <- as.character(cl$column)
   if (!is.character(items$ffcolname)) items$ffcolname <- as.character(items$ffcolname)
   if (!is.character(items$lookuptable)) items$lookuptable <- as.character(items$lookuptable)
   lookups <- subset(items, !is.na(items$lookuptable))
   lookups <- unique(lookups[, c("lookuptable", "ffcolname")])
 
+  # needed for the meta variable exception check below
+  meta_var_names <- unique(cl$column[! grepl(".", cl$column, fixed = TRUE)])
+
   str <- strsplit(cl$column, ".", fixed = TRUE)
-  str <- sapply(str, function(x) x[2])
+  # filters for all which were split by "." the rest is "NA"
+  str <- sapply(str, function(x) x[length(x)])
   cl$var <- str
   w <- cl$column %in% lookups$lookuptable
   cl$var[w] <- lookups$ffcolname[match(cl$column[w], lookups$lookuptable)]
 
-  for (i in names(data)[names(data) %in% cl$var]) {
-    lookup <- cl[grepl(paste0(form, ".", i, "$"), cl$column) |
+  for (name in names(data)[names(data) %in% cl$var]) {
+    # lookup for non-meta variables
+    lookup <- cl[grepl(paste0(form, ".", name, "$"), cl$column) |
                    (cl$var %in% names(data) & cl$var %in% lookups$ffcolname), ]
-    data[, paste0(i, ".factor")] <- factorize_secuTrial(data[, i], lookup)
-    data <- .move_column_after(data, paste0(i, ".factor"), i)
+    # exception for meta variables (for non meta variables the lookup should never be empty)
+    if (length(lookup$column) == 0 & name %in% meta_var_names) {
```

Comment:
why not use nrow(lookup) instead of length(lookup$column)?

Reply:
Fair point. I will change it.